File: app/src/main/java/in/mvpstarter/sample/injection/module/ActivityModule.java
package in.mvpstarter.sample.injection.module;

import android.app.Activity;
import android.content.Context;

import dagger.Module;
import dagger.Provides;
import in.mvpstarter.sample.injection.ActivityContext;

@Module
public class ActivityModule {

    private Activity mActivity;

    public ActivityModule(Activity activity) {
        mActivity = activity;
    }

    @Provides
    Activity provideActivity() {
        return mActivity;
    }

    @Provides
    @ActivityContext
    Context providesContext() {
        return mActivity;
    }
}

File: app/src/main/java/in/mvpstarter/sample/injection/module/ActivityModule.kt
package `in`.mvpstarter.sample.injection.module

import android.app.Activity
import android.content.Context

import dagger.Module
import dagger.Provides
import `in`.mvpstarter.sample.injection.ActivityContext

@Module
class ActivityModule(private val mActivity: Activity) {

    @Provides
    internal fun provideActivity(): Activity {
        return mActivity
    }

    @Provides
    @ActivityContext
    internal fun providesContext(): Context {
        return mActivity
    }
}
